Framework/Algorithms/src/ChangePulsetime2.cpp

```diff
@@ -0,0 +1,63 @@
+#include "MantidAlgorithms/ChangePulsetime2.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidDataObjects/EventWorkspace.h"
+#include "MantidKernel/ArrayProperty.h"
+#include "MantidKernel/System.h"
+
+namespace Mantid {
+namespace Algorithms {
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(ChangePulsetime2)
+
+using namespace Mantid::Kernel;
+using namespace Mantid::API;
+using namespace Mantid::DataObjects;
+using std::size_t;
+
+//----------------------------------------------------------------------------------------------
+/** Initialize the algorithm's properties.
+ */
+void ChangePulsetime2::init() {
+  declareIndexProperty<EventWorkspace>("InputWorkspace");
```

Comment:
I have discussed this with @LamarMoore, but would like a wider discussion about having implicit defaults for the index type in the declaration @martyngigg. I think I would prefer that the index type had to be explicitly declared (thus thought about). Workspace Indexes are great for internal use (development), but can lead to flakey scripts.